### ISSUE TITLE
[Flaking Test] [sig-auth] integration-master	TestMultiWebhookAuthzConfig

### ISSUE THREAD

**Issue by pacoxu** [kubernetes/kubernetes#124107]
### Which jobs are flaking?

- ci-kubernetes-integration-master

### Which tests are flaking?

--- FAIL: TestMultiWebhookAuthzConfig (9.23s)
k8s.io/kubernetes/test/integration/auth.auth

### Since when has it been flaking?

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1770274294463467520
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1773485209702895616 (03-20)

I cannot search for previous flake. 

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#integration-master

### Reason for failure (if possible)

```
    authz_config_test.go:669: failure timestamp changed from reload1Metrics.reloadFailure 2024-03-28 23:27:10.466200351 +0000 UTC to 2024-03-28 23:27:10.953506469 +0000 UTC unexpectedly
```

### Anything else we need to know?

https://github.com/kubernetes/kubernetes/blame/f4e246bc93ffb68b33ed67c7896c379efa4207e7/test/integration/auth/authz_config_test.go#L124 
updated recently

### Relevant SIG(s)

/sig auth
/cc @liggitt

**Comment by liggitt** [comment]
Updated the description, the actual cause of the flake is an unexpected change in the file reload timestamp

**Comment by liggitt** [comment]
interestingly, https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=k8s.io%2Fkubernetes%2Ftest%2Fintegration%2Fauth.auth doesn't show any failures of this test for this reason (there are a few failures on a PR that broke a bunch of integration tests for a different reason)

**Comment by pacoxu** [comment]
> Fix open in #124111, switched the file write to an atomic write

No flake after 03-29. Thanks.
